README.md

```diff
@@ -159,6 +161,9 @@ This tool currently is only useful for packing assets.
  * `rafx-api` - Rendering API abstraction layer.
    * Vulkan backend for windows/linux
    * Metal backend for macOS/iOS
+ * `rafx-visibility` - Retained visibility API.
```

Comment:
Honestly I'm not sure it makes a lot of sense to split the visibility code across two crates. Rafx started out with more crates (including a Rafx-visibility crate), but I ultimately decided it was not helpful and added more maintenance/complexity. (Nothing major or anything, but it was a constant cost that benefited no one.) I also think it's more friendly and approachable to users to understand the crates as a linear progression - i.e. a layer cake rather than a tree.

It's true that separating crates is *technically* more flexible, but I'm not sure who would actually use rafx-visibility but not rafx-framework. I'm willing to try it the way you have it now, but I wanted to mention it because I did already try it before and the experience made me want to have fewer crates.

Reply:
Even if we agree to merge the crates, I'm happy for us to put this in as it is and change it later.

Reply:
I have three at least three reasons why I'd like to keep it separate.

1. I feel somewhat more confident in the low-level API than I do in something higher-level with ref-counted smart pointers. I already ran into a couple of thorny problems implementing the `VisibilityRegion`, so keeping this separated is a way of insulating each side from the churn of the other side (assuming it's not along that shared boundary API)
2. I can foresee the lower-level crate getting progressively more complex (in terms of lines of code) to handle things like occlusion culling or hierarchical culling, whereas I wouldn't expect that to occur on the higher-level code since it's mostly providing the convenience of a more ergonomic interface, vs. functionality.
3. Where the higher-level code _does_ introduce new functionality, it's unrelated to the core problem of visibility -- e.g. associating visibility objects with render features.

Reply:
We could keep the two as separate modules within a single crate. But I'm fine either way. This is not hard to switch if we change our mind later :)